libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -530,6 +530,48 @@ void AP_Mission::write_home_to_storage()
     write_cmd_to_storage(0,home_cmd);
 }
 
+MAV_MISSION_RESULT AP_Mission::sanity_check_params(const mavlink_mission_item_int_t& packet) {
+    uint8_t nan_mask;
+    switch (packet.command) {
+        case MAV_CMD_NAV_WAYPOINT:
+            nan_mask = ~(1 << 3); // param 4 can be nan
+            break;
+        case MAV_CMD_NAV_LAND:
+            nan_mask = ~(1 << 3); // param 4 can be nan
+            break;
+        case MAV_CMD_NAV_TAKEOFF:
+            nan_mask = ~(1 << 3); // param 4 can be nan
+            break;
+        case MAV_CMD_NAV_VTOL_TAKEOFF:
+            nan_mask = ~(1 << 3); // param 4 can be nan
+            break;
+        case MAV_CMD_NAV_VTOL_LAND:
+            nan_mask = ~((1 << 2) || (1 << 3)); // param 3 and 4 can be nan
```

Comment:
This should be a `|`, not an `||`. I'll make a commit to fix it.

Reply:
good catch!